hybrid/pkg/server/server.go

```diff
@@ -0,0 +1,280 @@
+	selectors := []string{}
+	canReattest := true
+	for _, interceptor := range interceptors {
+		if !interceptor.CanReattest()[0] {
+			attested, err := agentstore.IsAttested(context.Background(), p.store, interceptor.SpiffeID())
+			canReattest = false
+			switch {
+			case err != nil:
+				return err
+			case attested:
+				return status.Error(codes.PermissionDenied, "attestation data has already been used to attest an agent")
+			default:
+			}
+		}
+		if len(p.interceptor.SpiffeID()) == 0 {
+			p.interceptor.SetSpiffeID(interceptor.SpiffeID())
+		}
+		selectors = append(selectors, interceptor.CombinedSelectors()...)
+	}
+
+	return p.interceptor.Stream().Send(&nodeattestorv1.AttestResponse{
+		Response: &nodeattestorv1.AttestResponse_AgentAttributes{
+			AgentAttributes: &nodeattestorv1.AgentAttributes{
+				CanReattest:    canReattest,
+				SpiffeId:       p.interceptor.SpiffeID(),
+				SelectorValues: selectors,
+			},
+		},
+	})
+}
+
+func (p *HybridPluginServer) Configure(ctx context.Context, req *configv1.ConfigureRequest) (*configv1.ConfigureResponse, error) {
+	pluginData, _ := p.decodeStringAndTransformToAstNode(req.HclConfiguration)
+
+	pluginNames, pluginsData := p.parseReceivedData(pluginData)
+
+	p.pluginList, p.initStatus = p.initPlugins(pluginNames)
+
+	if len(p.pluginList) == 0 || p.initStatus != nil {
+		return nil, p.initStatus
+	}
+
+	for i := 0; i < len(p.pluginList); i++ {
+		elem := reflect.ValueOf(p.pluginList[i].Plugin)
+		req.HclConfiguration = pluginsData[p.pluginList[i].PluginName]
+
+		result := elem.MethodByName("Configure").Call([]reflect.Value{reflect.ValueOf(ctx), reflect.ValueOf(req)})
+		err := result[1]
+
+		if !err.IsNil() {
+			return &configv1.ConfigureResponse{}, status.Errorf(codes.Internal, "Error configuring one of the supplied plugins(%s): %s", p.pluginList[i].PluginName, err.Interface().(error))
+		}
+	}
+
+	if brokerHostServices := p.setBrokerHostServices(); brokerHostServices != nil {
+		return &configv1.ConfigureResponse{}, brokerHostServices
+	}
+
+	return &configv1.ConfigureResponse{}, nil
+}
+
+func (p *HybridPluginServer) decodeStringAndTransformToAstNode(hclData string) (common.Generics, error) {
+	var genericData common.GenericPluginSuper
+	if err := hcl.Decode(&genericData, hclData); err != nil {
+		errorString := fmt.Sprintf("%v", err)
+		return nil, status.Errorf(codes.Internal, "Could not decode HCL config. The error was %v.", errorString)
+	}
+
+	var data bytes.Buffer
+	printer.DefaultConfig.Fprint(&data, genericData.Plugins)
+
+	var astNodeData common.Generics
+
+	if err := hcl.Decode(&astNodeData, data.String()); err != nil {
+		errorString := fmt.Sprintf("%v", err)
+		return nil, status.Errorf(codes.Internal, "Could not decode HCL config. The error was %v.", errorString)
+	}
+
+	return astNodeData, nil
+}
+
+func (p *HybridPluginServer) parseReceivedData(data common.Generics) ([]string, map[string]string) {
```

Comment:
It's because I'm adding a configuration inside the expected configuration. So when I'm getting the config, it comes like this:

{
 blablabla : { bla}
}

then I have to remove the { and } from both start and end